python/pyspark/ml/classification.py

```diff
@@ -1117,6 +1137,56 @@ def getBlockSize(self):
         """
         return self.getOrDefault(self.blockSize)
 
+    @since("2.0.0")
+    def setStepSize(self, value):
+        """
+        Sets the value of :py:attr:`stepSize`.
+        """
+        return self._set(stepSize=value)
+
+    @since("2.0.0")
+    def getStepSize(self):
+        """
+        Gets the value of stepSize or its default value.
+        """
+        return self.getOrDefault(self.stepSize)
+
+    @since("2.0.0")
+    def setSolver(self, value):
+        """
+        Sets the value of :py:attr:`solver`.
+        """
+        return self._set(solver=value)
+
+    @since("2.0.0")
+    def getSolver(self):
+        """
+        Gets the value of solver or its default value.
+        """
+        return self.getOrDefault(self.solver)
+
+    @property
+    @since("2.0.0")
+    def getOptimizer(self):
```

Comment:
Thank you for bringing that issue, @MLnick. `getOptimizer` is there to let the user know explicitly which optimizer is used. The reason why I did not implement `setOpmizer` is because minibatch gradient descent in not very efficient in Spark and can be used only in very specific cases. So the default choice for Spark is L-BFGS. However, if we label `setOpmizer` as `expertParam` then it is OK to have it. I believe that we should proceed as suggested by @yanboliang